mkosi

```diff
@@ -1228,6 +1255,58 @@ def patch_root_uuid(args, loopdev, root_hash):
         subprocess.run(["sfdisk", "--part-uuid", loopdev, str(args.root_partno), str(u)],
                        check=True)
 
+def install_unified_kernel(args, workspace, run_build_script, root_hash):
+
+    # Iterates through all kernel versions included in the image and
+    # generates a combined kernel+initrd+cmdline+osrelease EFI file
+    # from it and places it in the /EFI/Linux directory of the
+    # ESP. sd-boot iterates through them and shows them in the
+    # menu. These "unified" single-file images have the benefit that
+    # they can be signed like normal EFI binaries, and can encode
+    # everything necessary to boot a specific root device, including
+    # the root hash.
+
+    if not args.bootable:
+        return
+
+    if args.distribution != Distribution.fedora:
+        return
```

Comment:
hmm, i am a bit conservative on adding abstraction code without actually working on at least two backends... Hence I just hacked this up as it is, and then leave the abstraction work to the one who actually adds archlinux support...

(btw, I think it would be good if Arch's mkinitcpio could natively support a `--uefi` switch or so, the way dracut does. I think think objcopy thing is something that better belongs in the initrd implementation than in mkosi. but anyway, this is something for the arch maintainers to figure out)